crates/chain/src/tx_graph.rs

```diff
@@ -205,6 +205,26 @@ impl<A> TxGraph<A> {
         self.get_tx_node(txid).map(|n| n.tx)
     }
 
+    /// Get a transaction by txid. This only returns `Some` for full, unconfirmed transactions.
+    pub fn get_unconfirmed_tx<'g, C>(
+        &'g self,
+        txid: Txid,
+        chain: &'g C,
+        chain_tip: BlockId,
+    ) -> Result<Option<&'g Transaction>, C::Error>
+    where
+        A: Anchor,
+        C: ChainOracle,
+    {
+        let tx_node = self.get_tx_node(txid).expect("tx must exist");
```

Comment:
Given that the intention of this method is to be a public API, we should not panic if tx does not exist, since the caller provides the `txid`.